app/controllers/concerns/vet360/writeable.rb

```diff
@@ -67,5 +67,14 @@ def render_new_transaction!(type, response)
 
       render json: transaction, serializer: AsyncTransaction::BaseSerializer
     end
+
+    def request_is_delete?
+      request.request_method == 'DELETE'
+    end
+
+    def delete_safety_catch(params)
```

Comment:
When I read through each one of the controller actions, seeing `delete_safety_catch` seems out of place. It feels like you are removing a bunch of params.  No?

Perhaps change the name of`delete_safety_catch` to something like adjust(telephone_params ) or sanitize(telephone_params ), adapt(telephone_params ), or set_(telephone_params ), or something in that vein.

`delete_safety_catch` just seems like you are in fact deleting the params. When you are really not, you are just customizing one of the values in the params.

Reply:
In my opinion, splitting this into two methods like:

`add_effective_end_date(params)`
and
`strip_effective_end_date(params)`

would more clearly express the two actions here, remove the need for the `is_delete` check, and make reading the controller methods a little more clear as well.

Reply:
✅